annotation-tests/src/integration-test/java/eu/europeana/annotation/tests/web/AnnotationSearchIT.java

```diff
@@ -167,12 +170,38 @@ public void testSearchAnyAnnotationMinimalProfile() throws Exception {
     createAnnotationDataSet();
     // first page
     AnnotationPage annPg =
-        searchAnnotationsAddQueryField("*", null, null, null, null, SearchProfiles.MINIMAL, null);
+        searchAnnotationsAddQueryField("*", null, null, null, null, SearchProfiles.MINIMAL.toString(), null);
     assertNotNull(annPg, "AnnotationPage must not be null");
     // there might be old annotations of failing tests in the database
     assertEquals(annPg.getCurrentPage(), 0);
   }
 
+  @Test
+  public void searchWithFacetsBadRequests() throws Exception {
+    String url = AnnotationTestUtils.buildUrl("*:*", null, null, null, null, null, null, "minimal, facets, standard", null);
+    ResultActions mockMvcResult = mockMvc.perform(get(url));
+    assertEquals(HttpStatus.BAD_REQUEST, HttpStatus.valueOf(mockMvcResult.andReturn().getResponse().getStatus()));
+
+    url = AnnotationTestUtils.buildUrl("*:*", null, null, null, null, null, null, "minimal, standard", null);
+    mockMvcResult = mockMvc.perform(get(url));
+    assertEquals(HttpStatus.BAD_REQUEST, HttpStatus.valueOf(mockMvcResult.andReturn().getResponse().getStatus()));
+  }
+  
+  @Test
+  public void searchWithFacets() throws Exception {
+    createAnnotationDataSet();
+
+    String[] facets = {"motivation", "generator_uri"};
+    AnnotationPage annPg = searchAnnotations("*:*", null, facets, null, null, null, null, "minimal", null);
+    assertNotNull(annPg, "AnnotationPage must not be null");
+    assertNull(annPg.getFacets());
+
+    annPg = searchAnnotations("*:*", null, facets, null, null, null, null, "minimal, facets", null);
```

Comment:
there should be another test case for profile=debug in which case the default standard profile should be used for serialization. The presence of the motivation field can be used to verify if the standard serialization was used

Reply:
Added tests for profile=debug and profile=minimal,factes,debug. The search with different profiles is done in one test, no need for an overhead of creating 21 annotations for each test, just because the search profile changed.